add regex matching

luckman212 opened the pull request:
replaces simple string based matching with more powerful regex-based one

luckman212 commented:
@tgrosinger looks like the CI build is failing due to a hash mismatch on the API, maybe you have an older version referenced?

tgrosinger commented:
I do not think we should revert the stripping of `.md`. The paths used should match those displayed in the file tree in Obsidian, which hides `.md`.

tgrosinger commented:
> I do not think we should revert the stripping of `.md`. The paths used should match those displayed in the file tree in Obsidian, which hides `.md`.

Ah okay. I should have read the first commit before commenting this. If I understand correctly, if `.md` is not stripped, it will only have an effect if the user has `$` in their regex, correct?

tgrosinger commented:
> @tgrosinger looks like the CI build is failing due to a hash mismatch on the API, maybe you have an older version referenced?

Fixed, just rebase your changes on master to resolve.

luckman212 commented:
Thanks for looking @tgrosinger 👍 

At first, I was just stripping `.md` if it occurred at the very end of the filename, so a note named `Obsidian.md Notes` would not get mangled into `Obsidian Notes`.

As of my last commit, the filename is no longer stripped under any circumstance. The regex matching is done on the literal filepath e.g. `daily/2021-04-15.md` **would** be excluded by patterns `^daily.*` or `202[0-9].md$` but **not** `daily/[0-9]{4}-[0-9]{2}-[0-9]{2}$`

Do you agree with this ?

tgrosinger commented:
> Do you agree with this ?

Yes, I think so. Can you confirm that just `daily` would also match the file in your example?

tgrosinger commented on main.ts:
```diff
@@ -211,12 +211,12 @@ export default class RecentFilesPlugin extends Plugin {
 
   public readonly shouldAddFile = (file: FilePath): boolean => {
-    const filename = file.path.replace(new RegExp('.md$'), '');
+    // const filename = file.path.replace(new RegExp('.md$'), '');
```
Let's just remove this line instead of commenting it out. Ideally this change would be part of the previous commit rather than its own commit.

tgrosinger commented on main.ts:
```diff
@@ -211,12 +211,12 @@ export default class RecentFilesPlugin extends Plugin {
       try {
         const re = new RegExp(pattern);
-        const doesMatch: boolean = re.test(filename);
+        const doesMatch: boolean = re.test(file.path);
```
Ideally this change would be part of the previous commit rather than its own commit.

luckman212 commented:
@tgrosinger Yes, `daily` by itself would also match. I'll remove the commented line & rebase my fork...

tgrosinger commented:
This looks awesome, thank you! I'll test it out and get it merged soon 🙂

luckman212 commented:
@tgrosinger I pushed, squashed, rebased, pushed again, fixed something else, pushed and squashed some more. I don't like git. Hope this is all ok now.

tgrosinger commented:
Merged! Thank you
